src/components/Hand/Hand.tsx

```diff
@@ -3,12 +3,14 @@ import { GameStateContext } from "../GameStateProvider/GameStateProvider";
 import clsx from "clsx";
 import { useDrag, useDrop } from "react-dnd";
 import { Card, Hand } from "../../constants";
+import { range, shuffle } from "lodash";
 
 const BID_DRAG_TYPE = "__bid";
 
 const HandDisplay = ({ hand, index }: { hand: Hand; index: number }) => {
-  const { hands, swapBids, showCards } = useContext(GameStateContext);
+  const { hands, swapBids, showCards, numRevealedCards, numCardsPerHand } = useContext(GameStateContext);
   const [fadeAnimation, setFadeAnimation] = useState<boolean>(false);
+  const [cardRevealOrder, setCardRevealOrder] = useState<number[]>(() => shuffle(range(numCardsPerHand)));
```

Comment:
FYI (and I will fix this dw), using "derived state" like this is a bit of an antipattern in React. Instead we should initialize this at the same time as we initialize the hands, and then each hand gets a `displayOrder` field in addition to `cards`, `key`, and `bid`.